java-checks/src/main/java/org/sonar/java/symexecengine/DataFlowVisitor.java

```diff
@@ -104,7 +130,11 @@ public void visitTryStatement(TryStatementTree tree) {
     executionState = blockES;
     scan(tree.block());
     scan(tree.resources());
-    handleResources(tree.resources());
+    for (VariableTree resource : tree.resources()) {
+      for (SymbolicValue value : executionState.getValues(resource.symbol())) {
+        check.onTryResourceClosed(executionState, value);
```

Comment:
![New](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/new.png) ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Method 'SymbolicExecutionCheck.onTryResourceClosed(...)' is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)